File: app/src/main/java/com/satsumasoftware/timetable/db/ExamsSchema.java
package com.satsumasoftware.timetable.db;

import android.provider.BaseColumns;

import com.satsumasoftware.timetable.db.util.SchemaUtilsKt;

/**
 * The schema for the 'exams' table, containing constants for the column names and an SQLite create
 * statement.
 *
 * @see com.satsumasoftware.timetable.framework.Exam
 */
public final class ExamsSchema implements BaseColumns {

    public static final String TABLE_NAME = "exams";
    public static final String COL_TIMETABLE_ID = "timetable_id";
    public static final String COL_SUBJECT_ID = "subject_id";
    public static final String COL_MODULE = "module";
    public static final String COL_DATE_DAY_OF_MONTH = "date_day_of_month";
    public static final String COL_DATE_MONTH = "date_month";
    public static final String COL_DATE_YEAR = "date_year";
    public static final String COL_START_TIME_HRS = "start_time_hrs";
    public static final String COL_START_TIME_MINS = "start_time_mins";
    public static final String COL_DURATION = "duration";
    public static final String COL_SEAT = "seat";
    public static final String COL_ROOM = "room";
    public static final String COL_IS_RESIT = "is_resit";

    /**
     * An SQLite statement which creates the 'exams' table upon execution.
     *
     * @see TimetableDbHelper
     */
    static final String SQL_CREATE = "CREATE TABLE " + TABLE_NAME + "( " +
            _ID + SchemaUtilsKt.INTEGER_TYPE + SchemaUtilsKt.PRIMARY_KEY_AUTOINCREMENT + SchemaUtilsKt.COMMA_SEP +
            COL_TIMETABLE_ID + SchemaUtilsKt.INTEGER_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_SUBJECT_ID + SchemaUtilsKt.INTEGER_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_MODULE + SchemaUtilsKt.TEXT_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_DATE_DAY_OF_MONTH + SchemaUtilsKt.INTEGER_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_DATE_MONTH + SchemaUtilsKt.INTEGER_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_DATE_YEAR + SchemaUtilsKt.INTEGER_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_START_TIME_HRS + SchemaUtilsKt.INTEGER_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_START_TIME_MINS + SchemaUtilsKt.INTEGER_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_DURATION + SchemaUtilsKt.INTEGER_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_SEAT + SchemaUtilsKt.TEXT_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_ROOM + SchemaUtilsKt.TEXT_TYPE + SchemaUtilsKt.COMMA_SEP +
            COL_IS_RESIT + SchemaUtilsKt.INTEGER_TYPE +
            " )";

}

File: app/src/main/java/co/timetableapp/data/schema/ExamsSchema.kt
package co.timetableapp.data.schema

import android.provider.BaseColumns

/**
 * The schema for the 'exams' table, containing constants for the column names and an SQLite create
 * statement.
 *
 * @see co.timetableapp.model.Exam
 */
object ExamsSchema : BaseColumns {

    const val TABLE_NAME = "exams"
    const val _ID = BaseColumns._ID
    const val COL_TIMETABLE_ID = "timetable_id"
    const val COL_SUBJECT_ID = "subject_id"
    const val COL_MODULE = "module"
    const val COL_DATE_DAY_OF_MONTH = "date_day_of_month"
    const val COL_DATE_MONTH = "date_month"
    const val COL_DATE_YEAR = "date_year"
    const val COL_START_TIME_HRS = "start_time_hrs"
    const val COL_START_TIME_MINS = "start_time_mins"
    const val COL_DURATION = "duration"
    const val COL_SEAT = "seat"
    const val COL_ROOM = "room"
    const val COL_IS_RESIT = "is_resit"

    /**
     * An SQLite statement which creates the 'exams' table upon execution.
     *
     * @see co.timetableapp.data.TimetableDbHelper
     */
    internal const val SQL_CREATE = "CREATE TABLE " + TABLE_NAME + "( " +
            BaseColumns._ID + INTEGER_TYPE + PRIMARY_KEY_AUTOINCREMENT + COMMA_SEP +
            COL_TIMETABLE_ID + INTEGER_TYPE + COMMA_SEP +
            COL_SUBJECT_ID + INTEGER_TYPE + COMMA_SEP +
            COL_MODULE + TEXT_TYPE + COMMA_SEP +
            COL_DATE_DAY_OF_MONTH + INTEGER_TYPE + COMMA_SEP +
            COL_DATE_MONTH + INTEGER_TYPE + COMMA_SEP +
            COL_DATE_YEAR + INTEGER_TYPE + COMMA_SEP +
            COL_START_TIME_HRS + INTEGER_TYPE + COMMA_SEP +
            COL_START_TIME_MINS + INTEGER_TYPE + COMMA_SEP +
            COL_DURATION + INTEGER_TYPE + COMMA_SEP +
            COL_SEAT + TEXT_TYPE + COMMA_SEP +
            COL_ROOM + TEXT_TYPE + COMMA_SEP +
            COL_IS_RESIT + INTEGER_TYPE +
            " )"

}
